oxy/poc_translator.py

```diff
@@ -0,0 +1,73 @@
+from ruamel.yaml import YAML
+from functools import reduce
+
+
+yaml: YAML = YAML(typ="safe")
+
+'''
+pub fn decode_accumulator_status(data: &[u8]) -> HashMap<u8, f32> {
+    let mut result = HashMap::new();
+    result.insert(1, (pd::big_endian(&data[0..2], 8) as f32) / 10.0);
+    result.insert(
+        2,
+        pd::twos_comp(pd::big_endian(&data[2..4], 8) as u32, 16) as f32 / 10.0,
+    );
+    result.insert(3, pd::big_endian(&data[4..6], 8) as f32);
+    result.insert(4, data[6] as f32);
+    result.insert(5, data[7] as f32);
+    result
+}
+'''
+out_string: str = ""
+data = yaml.load(open("mapping.yaml"))[1]
```

Comment:
So per the nature of YAML and kinda this general application, I think that you should really think about the data structures going into this. The problem feels a lot like it is parsing a bunch of very similar handlers that are all _slightly_ different, meaning you can probably have a class or something per attribute of the handler. Probably also want to create a "CAN message" class that contains all the attributes associated with a CAN message found in the YAML and then use that to pass to a general decoder.